### PR TITLE
Fix IllegalAccessException when comparing 2 instances of class has private fields

### PR DESCRIPTION
**Pull request by phucnh**
Fix an issue https://github.com/scalatest/scalatest/issues/1788

When comparing two instances of class which contains private fields, `IllegalAccessException: cannot access a member with modifiers "private"` occurs at ObjectMeta.
It caused because `Differ` try to access to the instance's value for generating differences analysis message.

I fixed this problem by setting the field accessible to true temporarily.

### CODE REVIEW

**Comment by artimasites** [comment]
@cla-bot[bot] check

File: jvm/scalactic-test/src/test/scala/org/scalactic/source/ObjectMetaSpec.scala [hunk 1]
```diff
@@ -19,21 +19,23 @@ import org.scalatest._
 
 class ObjectMetaSpec extends funspec.AnyFunSpec with matchers.should.Matchers {
 
-  case class Person(name: String, age: Int) {
+  case class Person(name: String, private val age: Int) {
     val otherField = "test other field"
+    val privateField = "test private field"
```
Comment (cheeseng) [inline jvm/scalactic-test/src/test/scala/org/scalactic/source/ObjectMetaSpec.scala:24]: @phucnh  Er, do you actually intended to use a private val here?

Comment (phucnh) [reply jvm/scalactic-test/src/test/scala/org/scalactic/source/ObjectMetaSpec.scala]: @cheeseng Thank you for reviewing.
I have fixed https://github.com/scalatest/scalatest/pull/1834/commits/bb6a1e0bdd3a8b18e8cbb3efbc263d597ad785fe

**Comment by cheeseng** [comment]
@phucnh There's a test on ObjectMetaSpec that is failing when built under Dotty, I have submitted the following PR to your branch to fix it: 

https://github.com/phucnh/scalatest/pull/1

It seems like the behavior in Dotty is preferred one as compared to Scala 2.

**Comment by phucnh** [comment]
@cheeseng Thank you so much for fixing the failed test :bow:
I have merged your pull request

**Pull request merged**